### PR TITLE
Update ocamlformat to 0.18.0

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

**Comment by mnxn** [comment]
When I last tried to upgrade gen_js_api to 1.0.7, I got errors with our existing bindings: https://github.com/LexiFi/gen_js_api/issues/148

I'll wait on the CI, but it's possible 1.0.8 has the same problems.

**Comment by smorimoto** [comment]
Oh, indeed. It seems we need to fix it first.

**Comment by mnxn** [comment]
@smorimoto: I updated gen_js_api to 1.0.8, can you rebase this PR on top of master please?

**Comment by smorimoto** [comment]
@mnxn Sure!

File: .ocamlformat [hunk 1]
```diff
@@ -1,4 +1,5 @@
-version=0.15.0
+version=0.18.0
+profile=conventional
```
Comment (smorimoto) [inline .ocamlformat:2]: See https://github.com/ocaml/dune/pull/4351

**Review by mnxn — APPROVED** [review]
Looks good.

I think we can re-enable ocamlformat in CI now. It was disabled in https://github.com/ocamllabs/vscode-ocaml-platform/pull/570. Should that be done in this PR or a new one?

**Comment by smorimoto** [comment]
> I think we can re-enable ocamlformat in CI now. It was disabled in #570. Should that be done in this PR or a new one?

I will do that in a new one.

**Pull request merged**